src/hotspot/cpu/riscv/templateInterpreterGenerator_riscv.cpp

```diff
@@ -716,10 +702,12 @@ void TemplateInterpreterGenerator::lock_method() {
   }
 
   // add space for monitor & lock
+  __ check_extended_sp();
   __ add(sp, sp, - entry_size); // add space for a monitor entry
   __ add(esp, esp, - entry_size);
-  __ mv(t0, esp);
-  __ sd(t0, monitor_block_top);  // set new monitor block top
+  __ mv(t0, sp);
+  __ sd(t0, Address(fp, frame::interpreter_frame_extended_sp_offset * wordSize));
```

Comment:
Why not store sp directly here? Looks like use of t0 here is not necessary.

Reply:
Yes, we can store sp here directly. The same issue in other places was fixed too.